src/v/cluster/health_monitor_frontend.cc

```diff
@@ -147,4 +148,13 @@ ss::future<bool> health_monitor_frontend::does_raft0_have_leader() {
       [](health_monitor_backend& be) { return be.does_raft0_have_leader(); });
 }
 
+ss::future<> health_monitor_frontend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Frontend");
+    // updated_disk_health_cache will force a refresh on the backend with
+    // timeouts which will indirectly "reset" internal state on the backend
+    co_await update_disk_health_cache(force_refresh::yes);
```

Comment:
This is an alternative to wiping state. See this thread for details
https://github.com/redpanda-data/redpanda/pull/9282#discussion_r1136043149

Reply:
This entire PR LGTM, curious if others have objections to this impl over 9282

Reply:
this is a little bit counter intuitive. we want to refresh whole state but calling `refresh_disk_health_cache`. Can we have a method that will simply refresh the health monitor backend cache ?

Reply:
Sure I'll make a method to refresh the backend, but I do not see how it will be different from `update_disk_health_cache`.

Reply:
Done.